### PR TITLE
FC-0057: implement pin/unpin APIs

### LINKED ISSUE #6: Migrate the /pins API endpoints
Estimated time: 3 days

### PR DESCRIPTION
**Pull request by Faraz32123**
- this PR contains pin/unpin APIs and it's tests
close #6

### CODE REVIEW

**Comment by Faraz32123** [comment]
created a separate issue for pin/unpin APIs tests.
https://github.com/edly-io/forum/issues/40

File: forum/utils.py [hunk 1]
```diff
@@ -32,3 +44,93 @@ def handle_proxy_requests(request: HttpRequest, suffix: str, method: str) -> Res
         headers=request_headers,
         timeout=5.0,
     )
+
+
+def validate_thread_and_user(
```
Comment (taimoor-ahmed-1) [inline forum/utils.py:49]: can we move this code to `models/model_utils` to place mongodb specific code over there?

**Review by taimoor-ahmed-1 — CHANGES_REQUESTED** [review]

File: forum/utils.py (hunk 1 above)
Comment (Faraz32123) [reply forum/utils.py:49]: yes sure, we can.

**Review by Ali-Salman29 — APPROVED** [review]
Remove this empty file `tests/test_views/test_pins.py`.

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM

**Pull request merged**